src/app/public/modules/wait/wait-adapter.service.ts

```diff
@@ -21,13 +36,95 @@ export class SkyWaitAdapterService {
     waitEl: ElementRef,
     isFullPage: boolean,
     isWaiting: boolean,
-    isNonBlocking = false
-  ): void {
+    id: string,
+    waitCmp: SkyWaitComponent
+  ) {
     let busyEl = isFullPage ? document.body : waitEl.nativeElement.parentElement;
     let state = isWaiting ? 'true' : undefined;
 
-    if (!isNonBlocking) {
+    if (!(waitCmp && waitCmp.isNonBlocking)) {
       this.renderer.setElementAttribute(busyEl, 'aria-busy', state);
+
+      // Remove focus from page when full page blocking wait
+      if (isFullPage || busyEl.contains(document.activeElement)) {
+        this.clearDocumentFocus();
```

Comment:
In the visual demo, if I trigger "full page" and then trigger the wait, this works perfectly. However, if I trigger the wait first, then switch it to "full page", focus still remains on the button. Is that just a side effect of the visual demo being weird? Is this a real-world scenario we should account for?

Reply:
I still don't think that is a valid case, but I should be able to account for it